From f660c4797eab05d82a61c8d3ad10ba26e1f85cf8 Mon Sep 17 00:00:00 2001
From: Thomas Gerbet <thomas.gerbet@enalean.com>
Date: Fri, 21 Apr 2023 14:18:00 +0200
Subject: [PATCH] Disable process default timeout when running the install and
 migration scripts

This is a followup to d6a3df3c9d251a3b707294c97ddc4f9be9e9877a where
those elements were missed.

Change-Id: Iebcfa218ba30c459e7c738a22889bcb2d8094926
---
 maintenance/migrateInstance.php     | 1 +
 maintenance/runForAll.php           | 2 +-
 src/ProcessStep/InstallInstance.php | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/extensions/TuleapWikiFarm/maintenance/migrateInstance.php b/extensions/TuleapWikiFarm/maintenance/migrateInstance.php
index b4b8762..4422afd 100644
--- a/extensions/TuleapWikiFarm/maintenance/migrateInstance.php
+++ b/extensions/TuleapWikiFarm/maintenance/migrateInstance.php
@@ -103,6 +103,7 @@ class MigrateInstance extends Maintenance {
 			[ '--quick' ],
 			[ '--sfr', $instance->getName() ]
 		) );
+		$process->setTimeout( null );
 		$process->run();
 		if ( $process->getExitCode() ) {
 			$this->fatalError( "Could not migrate database: " . $process->getErrorOutput() . "\n" );
diff --git a/extensions/TuleapWikiFarm/maintenance/runForAll.php b/extensions/TuleapWikiFarm/maintenance/runForAll.php
index ca9c2a7..fdfaaa9 100644
--- a/extensions/TuleapWikiFarm/maintenance/runForAll.php
+++ b/extensions/TuleapWikiFarm/maintenance/runForAll.php
@@ -37,7 +37,7 @@ class RunForAll extends Maintenance {
 				explode( ' ', $this->getOption( 'args' ) ),
 				[ '--sfr', $name ]
 			) );
-
+			$process->setTimeout( null );
 			$this->output( "Executing for $name\n" );
 			if ( $this->getOption( 'set-maintenance', false ) ) {
 				$this->setMaintenanceMode( $instance, $manager );
diff --git a/extensions/TuleapWikiFarm/src/ProcessStep/InstallInstance.php b/extensions/TuleapWikiFarm/src/ProcessStep/InstallInstance.php
index a8e14bb..4191f36 100644
--- a/extensions/TuleapWikiFarm/src/ProcessStep/InstallInstance.php
+++ b/extensions/TuleapWikiFarm/src/ProcessStep/InstallInstance.php
@@ -113,6 +113,7 @@ class InstallInstance implements IProcessStep {

 		// We must run this in isolation, as to not override globals, services...
 		$process = new Process( $processArgs );
+		$process->setTimeout( null );

 		$err = '';
 		$process->run( static function ( $type, $buffer ) use ( &$err ) {
--
2.40.0

